### PR TITLE
bpo-39080: Starred Expression's column offset fix when inside a CALL

### PR DESCRIPTION
**Pull request by lysnikolaou**
https://bugs.python.org/issue39080

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2019-12-17-21-45-36.bpo-39080.OrxEVS.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix a Starred Expressions's end_col_offset, when it is inside a Call object.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2019-12-17-21-45-36.bpo-39080.OrxEVS.rst]:
```suggestion
Fix the value of *end_col_offset* for Starred Expression AST nodes when they are among the elements in the *args* attribute of Call AST nodes.
```

**Comment by pablogsal** [comment]
Actually, this seems that does not only affect starred expressions. For example:

```
>>> val = "f(a,b,c)"
>>> val[ast.parse(val).body[0].value.args[2].end_col_offset]
')'
>>> val[ast.parse(val).body[0].value.args[0].end_col_offset]
','
>>> val[ast.parse(val).body[0].value.args[1].end_col_offset]
','
>>> val[ast.parse(val).body[0].value.args[2].end_col_offset]
')'
```

that seems odd.

**Comment by gvanrossum** [comment]
Possibly @ilevkivskyi would also be interested in hearing about this?

**Comment by pablogsal** [comment]
Nevermind, I imagine that is so you can do `expression[col_offset:end_col_offset]`.

**Comment by lysnikolaou** [comment]
> Actually, this seems that does not only affect starred expressions.

That's actually ok, because `end_col_offset` is always where the token ends + 1. In your examples `col_offset` of `a` is 2 and `end_col_offset` is 3, which is the intended behavior.

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
> That's actually ok, because `end_col_offset` is always where the token ends + 1. In your examples `col_offset` of `a` is 2 and `end_col_offset` is 3, which is the intended behavior.

Yup. Also the fix is consistent with extracting the values via `expression[col_offset:end_col_offset]`

**Review by ilevkivskyi — APPROVED** [review]
Thanks! This totally makes sense, with the current logic end position is the end position of the `*` symbol itself, which is wrong.

**Comment by ilevkivskyi** [comment]
This doesn't need backport to 3.7 as this is a new feature in 3.8

**Comment by lysnikolaou** [comment]
@ilevkivskyi A similar change I made in #17582 was considered a bugfix and was backported to 3.7

**Comment by pablogsal** [comment]
@lysnikolaou  In #17582 what changed was the `col_offset` that is accessible though Python AST objects, but `end_col_offset` and `end_lineno` was introduced in 3.8.

**Pull request merged**